addons/web_editor/models/ir_qweb.py

```diff
@@ -202,6 +204,21 @@ def attributes(self, record, field_name, options, values):
         attrs = super(Date, self).attributes(record, field_name, options, values)
         if options.get('inherit_branding'):
             attrs['data-oe-original'] = record[field_name]
+
+            if record._fields[field_name].type == 'datetime':
+                attrs = self.env['ir.qweb.field.datetime'].attributes(record, field_name, options, values)
+                attrs['data-oe-type'] = 'datetime'
+                return attrs
+
+            lg = self.env['res.lang']._lang_get(self.env.user.lang)
+            locale = babel.Locale.parse(lg.code)
```

Comment:
This breaks as soon as `self.env.user.lang` is unset, because the `lang` field is not required. Could you urgently revisit the whole patch to care for that situtation + fix all branches?
We've just broken qweb rendering on many production databases for the sake of fixing a corner case :-/

Reply:
Relevant part of the traceback:

```
File "./src/odoo/13.0/addons/web_editor/models/ir_qweb.py", line 248, in attributes
    locale = babel.Locale.parse(lg.code)
File "/usr/lib/python3/dist-packages/babel/core.py", line 265, in parse
    raise TypeError('Unexpected value for identifier: %r' % (identifier,))
TypeError: Unexpected value for identifier: False
```

Reply:
in method `odoo.tools.misc.format_date` that do the same thing, they do this to get the lang:

```
def get_lang(env, lang_code=False):
    """                                                                      
    Retrieve the first lang object installed, by checking the parameter lang_code,                              
    the context and then the company. If no lang is installed from those variables,
    fallback on the first lang installed in the system.
    :param str lang_code: the locale (i.e. en_US)                                                                            
    :return res.lang: the first lang found that is installed on the system.                      
    """                      
    langs = [code for code, _ in env['res.lang'].get_installed()]
    for code in [lang_code, env.context.get('lang'), env.user.company_id.partner_id.lang, langs[0]]:
        if code in langs:
            return env['res.lang']._lang_get(code)

def format_date(env, value, lang_code=False, date_format=False):
...
    lang = get_lang(env, lang_code)                                                                             
    locale = babel.Locale.parse(lang.code)
...
```

Reply:
@nle-odoo This will be fixed with https://github.com/odoo/odoo/pull/46383 by indeed using that `get_lang` method.

> We've just broken qweb rendering on many production databases for the sake of fixing a corner case :-/

After what we discussed, it seems to only impact users that are publishers, that have no lang set and [not in lower version than Odoo 13.0](https://github.com/odoo/odoo/pull/46383#pullrequestreview-365687342).